sql-plugin/src/main/scala/org/apache/spark/sql/rapids/GpuScalarSubquery.scala

```diff
@@ -66,8 +66,9 @@ case class GpuScalarSubquery(
       ExprId(0))
   }
 
-  override def columnarEval(batch: ColumnarBatch): Any = {
+  override def columnarEval(batch: ColumnarBatch): GpuColumnVector = {
     require(updated, s"$this has not finished")
-    GpuScalar(result, dataType)
+    GpuExpressionsUtils.resolveColumnVector(
```

Comment:
> This PR goes 99% of the way there, but we need to be sure that all of the binary and trinary ops are really covering everything. Unless @abellina did that as a part of this already. Then yes we should probably do it here.

I audited the code when I put this up and did it again. Any call to `columnarEvalAny` is checking that the result is a scalar or vector. But perhaps we can do better here, and we can extract the match statements to a higher level.

> We should also file a follow on issue with update any code that requires a lit, to see if we can get them to also support scalarSubQuery output.

Do you have an example for this? Do you mean to find code that normally is planned as having a `ScalarSubQuery` child and see if it can be optimized to invoke `columnarEvalAny`?

Reply:
I guess this is one example: https://github.com/NVIDIA/spark-rapids/pull/8333/files#diff-9f0f07939d25080e000ca9d939ac1443eef516868843d09ccca0d44b8666d651R146.

`GpuTimeAdd` was changed (really the reason this PR was prioritized) because the left child was a scalar subquery and returned a `GpuScalar`, and it wasn't ready to handle it. The fix was to turn the scalar into a column (this PR calls `columnarEval` so same effect).

`GpuTimeAdd` could support scalars on the LHS if we changed the code a bit, because I think we can add something to a scalar.. so I am not sure why it's not supported.